tmt/steps/prepare/install.py

```diff
@@ -10,6 +10,253 @@
 COPR_URL = 'https://copr.fedorainfracloud.org/coprs'
 
 
+class InstallBase(tmt.utils.Common):
+    """ Base class for installation implementations """
+
+    def __init__(self, parent, guest, package, pkg_mgr, data, plugin='dnf-plugins-core'):
```

Comment:
1. `pkg_mgr` - shouldn't this be an implementation detail of each class? I understand why the base class should be aware, but this parameter shouldn't probably "leak" into child classes' `__init__()` methods - I see no reason to inform `InstallRpmOstree` that it should use `rpm-ostree`.
2. can we find a better name for `data`? My feeling is there are way too many "data" blobs flying around, more expressive names would be better.
3. `plugin` takes some weird route: it's given to `__init__()`, set to `self.plugin`, and then used from outside of the class when calling `installer.enable_copr(..., plugin=installer.plugin, ...)`. So, either `plugin` is not necessary, since `self.plugin` is written once but never read, or it should be set once and then used by `enable_copr()` without passing it to this method again.
4. use of `yum` requires a bit too much info about the internals, I need to override both `pkg_mgr` and `plugin`. That suggests we might do better with a subclass of `InstallDnf`, `InstallYum`, which would call its super class' `__init__()` with proper package manager and plugin.

Reply:
1. - [ ] 

2. - [ ] 

3. - [ ]